defaults/helpers.go

```diff
@@ -18,3 +33,188 @@ func ociProfileConversionCheck(profile secprofile.Profile, entitlementID string)
+		// In case we get an error before seccomp struct is fully updated from decoded json, we empty it manually.
+		s.Linux.Seccomp = &specs.LinuxSeccomp{}
+	}
+
+	s.Linux.Seccomp = seccomp
+
+	return s
+}
+
+// We can't create a map to check set inclusion with specs.LinuxSeccompArg as key type in Golang so no O(n)
+func seccompArgsContains(syscallArgsSet, syscallArgsSubset []specs.LinuxSeccompArg) bool {
+	if len(syscallArgsSet) < len(syscallArgsSubset) ||
+		(len(syscallArgsSubset) == 0 && len(syscallArgsSet) != 0) {
+		return false
+	}
+
+	for _, syscallArgFromSubset := range syscallArgsSubset {
+		found := false
+
+		for _, syscallArgsFromSet := range syscallArgsSet {
+			if reflect.DeepEqual(syscallArgFromSubset, syscallArgsFromSet) {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return false
+		}
+	}
+
+	return true
+}
+
+func matchSeccompRule(seccompRule specs.LinuxSyscall, syscallName string, syscallArgs []specs.LinuxSeccompArg) bool {
+	for _, name := range seccompRule.Names {
+		if name == syscallName {
+			if seccompArgsContains(seccompRule.Args, syscallArgs) {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func seccompSyscallWithArgsBlocked(seccompProfile specs.LinuxSeccomp, syscallName types.Syscall, syscallArgs []specs.LinuxSeccompArg) bool {
+	syscallNameStr := string(syscallName)
+
+	blocked := seccompProfile.DefaultAction == specs.ActErrno
+
+	if blocked {
+		// For each rule in the seccomp profile, make sure that no whitelisting rule contains this syscall
+		for _, seccompRule := range seccompProfile.Syscalls {
+			if seccompRule.Action == specs.ActAllow {
+				// If we match a whitelisting rule containing this syscall and those arguments, syscall is not blocked
+				if matchSeccompRule(seccompRule, syscallNameStr, syscallArgs) {
+					return false
+				}
+			}
+		}
+	} else {
+		// For each rule in the seccomp profile, make sure that at least one blacklisting rule contain this syscall
+		for _, seccompRule := range seccompProfile.Syscalls {
+			if seccompRule.Action == specs.ActErrno {
+				// If we match a blacklisting rule containing this syscall and those arguments, syscall is blocked
+				if matchSeccompRule(seccompRule, syscallNameStr, syscallArgs) {
+					return true
+				}
+			}
+		}
+	}
+
+	return blocked
+}
+
+func seccompSyscallsWithArgsBlocked(seccompProfile specs.LinuxSeccomp, syscallsWithArgs map[types.Syscall][]specs.LinuxSeccompArg) bool {
+	for syscallName, syscallArgs := range syscallsWithArgs {
+		if !seccompSyscallWithArgsBlocked(seccompProfile, syscallName, syscallArgs) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func seccompSyscallsBlocked(seccompProfile specs.LinuxSeccomp, syscallNames []types.Syscall) bool {
+	for _, syscallName := range syscallNames {
+		if !seccompSyscallWithArgsBlocked(seccompProfile, syscallName, []specs.LinuxSeccompArg{}) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func capBlocked(linuxCaps specs.LinuxCapabilities, capability types.Capability) bool {
```

Comment:
I'm wondering if this should be configurable to also account for ambient caps?  Docker doesn't use them as of current, but it would be useful for the future / other workloads.

Reply:
[As discussed] that makes sense but for now I don't want false positive on Moby. I have to talk with the runc people to understand the tradeoffs, who might use ambient caps and why.